ReText/tab.py

```diff
@@ -444,15 +444,76 @@ def openSourceFile(self, fileToOpen):
 		"""Finds and opens the source file for link target fileToOpen.
 
 		When links like [test](test) are clicked, the file test.md is opened.
+		# TODO: broken objective? See TODO below.
 		It has to be located next to the current opened file.
 		Relative paths like [test](../test) or [test](folder/test) are also possible.
+
+		:param fileToOpen:	Link target
+							This can be a file of any extension or a path
+		:returns: 			Filename successfully handled to openFileWrapper
+							or 'noSuchFile' to stop further processing.
+		:rtype: 			str
 		"""
 		if self.fileName:
+			# Please give description of this condition.
+			# (e.g. "if internal link was clicked from within valid open document in a tab"?)
 			currentExt = splitext(self.fileName)[1]
 			basename, ext = splitext(fileToOpen)
 			if ext in ('.html', '') and exists(basename + currentExt):
+				# TODO: It seems this should catch markdown links like
+				#  [test](test) but than ext not in ('.html', '')
 				self.p.openFileWrapper(basename + currentExt)
 				return basename + currentExt
-		if exists(fileToOpen) and get_markup_for_file_name(fileToOpen, return_class=True):
+
+		if get_markup_for_file_name(fileToOpen, return_class=True):
+			if not QFile.exists(fileToOpen):
+				# ReText should only support to create markup-files, no plain
+				# text files. Thereby it facilitates writing cross-linked
+				# documentation.
+				# Consequently create-file functionality can be nested here.
+				if not self.createFileQuestion(fileToOpen):
+					# File still not exists
+					return 'noSuchFile'
+
+			# fileToOpen exists and has markup
 			self.p.openFileWrapper(fileToOpen)
 			return fileToOpen
+
+	def createFileQuestion(self, fileToCreate):
+		"""Ask user if file should be created and do so if answer is yes.
+
+		:returns: True if new file has been created, False otherwise
+		:rtype: bool
+		"""
+		buttonReply = QMessageBox.question(self, self.tr('Create missing file?'),
+			self.tr("The file '%s' does not exist.\n\nDo you want to create it?" % fileToCreate),
```

Comment:
Here and in other places: you should pass **unsubstituted** string to `tr()`. E.g. not
```python
self.tr("The file '%s' does not exist.\n\nDo you want to create it?" % fileToCreate)
```
but rather
```python
self.tr("The file '%s' does not exist.\n\nDo you want to create it?") % fileToCreate
```

Otherwise translating just won't work.